### ISSUE TITLE
Dynamic API Routes fail in production build

### ISSUE THREAD

**Issue by dbachrach** [zeit/next.js#7807]
# Bug report

## Describe the bug

Dynamic API routes (`/api/people/[id]`) fails in production builds (`yarn build` followed by `yarn start`). It returns an Internal Server Error with the following stack:

```
{ Error: Cannot find module for page: /api/people/4
    at pageNotFoundError (/Users/dustin/Desktop/api-routes-app/node_modules/next-server/dist/server/require.js:13:17)
    at Object.getPagePath (/Users/dustin/Desktop/api-routes-app/node_modules/next-server/dist/server/require.js:33:19)
    at Server.resolveApiRequest (/Users/dustin/Desktop/api-routes-app/node_modules/next-server/dist/server/next-server.js:200:26)
    at Server.handleApiRequest (/Users/dustin/Desktop/api-routes-app/node_modules/next-server/dist/server/next-server.js:181:43)
    at Object.fn (/Users/dustin/Desktop/api-routes-app/node_modules/next-server/dist/server/next-server.js:146:32)
    at /Users/dustin/Desktop/api-routes-app/node_modules/next-server/dist/server/router.js:20:36
    at Server.run (/Users/dustin/Desktop/api-routes-app/node_modules/next-server/dist/server/next-server.js:236:23)
    at Server.handleRequest (/Users/dustin/Desktop/api-routes-app/node_modules/next-server/dist/server/next-server.js:88:21)
    at Server.emit (events.js:182:13)
    at parserOnIncoming (_http_server.js:651:12) code: 'ENOENT' }
```

## To Reproduce

You can reproduce this with the nextjs example `api-routes`

```
yarn create next-app --example api-routes api-routes-app
cd api-routes-app
yarn install
yarn dev
```

Visiting http://localhost:3000/api/people/4 works. Stop the dev build and:

```
yarn build
yarn start
```

Visiting http://localhost:3000/api/people/4 yields `Internal Server Error`

## Expected behavior

Dynamic Api Routes work in dev and production builds.

## System information

- OS: macOS
- Browser: Chrome
- Version of Next.js: 9.0.0

**Comment by ijjk** [comment]
Hi, this should have been resolved in #7774. Can you try `next@ v9.0.1-canary.0` and confirm?

**Comment by dbachrach** [comment]
Confirmed this works on the canary. Thanks.
